op-node/rollup/derive/channel.go

```diff
@@ -121,11 +137,11 @@ func (ch *Channel) IsReady() bool {
 func (ch *Channel) Reader() io.Reader {
 	var readers []io.Reader
 	for i := uint64(0); i <= uint64(ch.endFrameNumber); i++ {
-		data, ok := ch.inputs[i]
+		frame, ok := ch.inputs[i]
 		if !ok {
 			panic("dev error in channel.Reader. Must be called after the channel is ready.")
 		}
-		readers = append(readers, bytes.NewBuffer(data))
+		readers = append(readers, bytes.NewBuffer(frame.Data))
```

Comment:
```suggestion
		readers = append(readers, bytes.NewReader(frame.Data))
```
We don't need the buffer functionality here I think, so maybe a regular bytes reader is better?